consensus/state.go

```diff
@@ -1470,7 +1470,22 @@ func (cs *ConsensusState) recordMetrics(height int64, block *types.Block) {
 			missingValidators++
 			missingValidatorsPower += val.VotingPower
 		}
+
+		_, privVal := cs.LastValidators.GetByAddress(cs.privValidator.GetPubKey().Address())
```

Comment:
I believe we need to wrap the whole thing in `if cs.privValidator != nil { .... }`